src/main/scala/com/snowplowanalytics/righttobeforgotten/ValueFilters.scala

```diff
@@ -0,0 +1,215 @@
+/**
+ * Copyright 2018 Snowplow Analytics
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.snowplowanalytics.righttobeforgotten
+
+import com.fasterxml.jackson.databind.{ObjectMapper, SerializationFeature}
+import com.fasterxml.jackson.databind.node.{ObjectNode, TextNode}
+import com.github.fge.jsonschema.core.report.ProcessingMessage
+import com.jayway.jsonpath.spi.json.JacksonJsonNodeJsonProvider
+import com.snowplowanalytics.iglu.client.repositories.{HttpRepositoryRef, RepositoryRefConfig}
+import org.json4s.jackson.JsonMethods._
+import org.apache.spark.sql.functions.udf
+import com.snowplowanalytics.snowplow.analytics.scalasdk.json.EventTransformer
+import com.snowplowanalytics.iglu.client.{Resolver, SchemaCriterion, SchemaKey, ValidatedNel}
+import com.snowplowanalytics.iglu.client.validation.ValidatableJValue
+import org.json4s.{JArray, JNothing, JObject, JValue}
+import scalaz.{Failure, Success}
+import com.jayway.jsonpath.{Configuration, JsonPath, Option => JOption}
+import org.json4s.jackson.JsonMethods
+import org.apache.commons.vfs2.VFS
+import scala.io.Source
+
+sealed trait ValueFilter {
+  val fieldName: String
+
+  /**
+   * Tests whether this filter matches this event
+   * @param event
+   * @return Boolean true or false
+   */
+  def matches(inputRows: JValue): Boolean
+}
+
+object ValueFilters {
+  val igluCentral    = RepositoryRefConfig("Iglu central", 0, List("com.snowplowanalytics"))
+  val httpRepository = HttpRepositoryRef(igluCentral, "http://iglucentral.com")
+  val resolver       = Resolver(500, List(httpRepository))
+
+  /**
+   * Create filters from removal criteria
+   * @param removalCriteriaLines
+   * @return either a list of strings containing the errors or a list of value filters
+   */
+  def createFiltersFromRemovalCriteria(removalCriteriaLines: List[String]): Either[List[String], List[ValueFilter]] =
+    removalCriteriaLines
+      .map(validateSingleCriterion)
+      .partition(_.isLeft) match {
+      case (failures, _) if (failures.nonEmpty) =>
+        Left(for (Left(s) <- failures) yield s)
+      case (_, filters) => Right(for (Right(f) <- filters) yield f)
+    }
+
+  /**
+   * Get removal criteria lines from file
+   * @param removalCriteriaFile
+   * @return
+   */
+  def getRemovalCriteriaLines(removalCriteriaFile: String): List[String] = {
```

Comment:
do we need a more typesafe wrapping here? what happens if the file doesn't exist?

Reply:
Not sure what you mean here.. Can you be more specific? What would you do? (and why)

Reply:
this function would give back an `Either[E, List[String]]` and the left would contain the failures cases e.g. the file doesn't exist